k-test/k-test-spec.ini

```diff
@@ -0,0 +1,56 @@
+[root]
+k: (#execute => #halt) ~> _
+schedule: BYZANTIUM
+gas: 1000000 => _
+balance: 0
+log: _
+refund: _
+storage: _:Map
+origStorage: _
+callDepth: CD
+requires:
+    andBool #range(0 <= CD < 1023)
+
+; callData is set in subspecs
+[ecrecT]
+statusCode: _ => EVMC_SUCCESS
++requires:
+    // Range
+    andBool #rangeAddress(      A1 )
+    andBool #rangeBytes( 32,    P1 )
+    andBool #rangeUInt( 8,      A2V )
+    andBool #rangeBytes( 32,    P3 )
+    andBool #rangeBytes( 32,    P4 )
+    andBool #rangeAddress(      A3 )
+
+    andBool ECREC_DATA ==K #encodeArgs( #bytes32(P1), #uint8(A2V), #bytes32(P3), #bytes32(P4) )
+    andBool A2 ==Int #symEcrec( ECREC_DATA )
+    andBool notBool #ecrecEmpty( ECREC_DATA )
+
+[ecrecT-false]
+output: _ => #encodeArgs( #bool(0) )
++requires:
+    andBool (A1 >=Int A2 orBool A2 >=Int A3 )
+
+; SPEC fails on old K. Requires abstraction of #symEcrec into fresh var in implication Z3 query.
+; #symEcrec must not have [smtlib], otherwise test is pointless.
+; function ecrecImplication(address a1, bytes32 p1, uint8 a2v, bytes32 p3, bytes32 p4, address a3)
+[ecrecT-false-implication]
+callData: #abiCallData("ecrecImplication", #address(A1), #bytes32(P1), #uint8(A2V), #bytes32(P3), #bytes32(P4), #address(A3))
+
+; SPEC fails on old K. Requires abstraction of #symEcrec into fresh var in constraint Z3 query.
+; function ecrecConstraint(address a1, bytes32 p1, uint8 a2v, bytes32 p3, bytes32 p4, address a3)
+[ecrecT-false-constraint]
+callData: #abiCallData("ecrecConstraint", #address(A1), #bytes32(P1), #uint8(A2V), #bytes32(P3), #bytes32(P4), #address(A3))
```

Comment:
these two specs will be proved by the old K if we put the smtlib attribute, right?

Reply:
Good question. that actually won't be enough. The argument of #symEcrec is WordStack, so we would have to also mark `Int ":" WordStack` with smtlib. And then we would have very long queries with little useful information, which will likely timeout.

This is actually a good example when we cannot use smtlib and need improved term abstraction.

Reply:
PR #94 opened: experiment with old K + smtlib attributes.